src/main/frontend/handler/editor.cljs

```diff
@@ -589,19 +571,24 @@
              input (gdom/getElement (state/get-edit-input-id))
              pos (cursor/pos input)
              [fst-block-text snd-block-text] (compute-fst-snd-block-text value pos)
-             insert-fn (match (mapv boolean [block-self? (seq fst-block-text) (seq snd-block-text)])
-                         [true _ _] insert-new-block-aux!
-                         [_ false true] insert-new-block-before-block-aux!
-                         [_ _ _] insert-new-block-aux!)]
+             insert-fn (cond
+                         block-self?
+                         insert-new-block-aux!
+
+                         (and (string/blank? fst-block-text) (not (string/blank? snd-block-text)))
+                         insert-new-block-before-block-aux!
+
+                         :else
+                         insert-new-block-aux!)]
          (insert-fn config block value
                     {:ok-handler
                      (fn [last-block]
-                       (edit-block! last-block 0 id)
-                       (clear-when-saved!))}))))
+                       (clear-when-saved!)
+                       (edit-block! last-block 0 id))}))))
    (state/set-editor-op! nil)))
 
 (defn api-insert-new-block!
```

Comment:
@tiensonqin This breaks the need for PR https://github.com/logseq/logseq/pull/4822. Any new way to achieve this?

Reply:
@llcc oops, added back with https://github.com/logseq/logseq/pull/4880/commits/025bff7a6ac6cc60585389a93eb88bdbd2b244e7.

Reply:
Cool. Then all those following places should also be addressed. 
https://github.com/logseq/logseq/search?q=%3Areuse-last-block%3F&type=

Reply:
@tiensonqin It seems that the current implementation replaces the last block no matter whether it is empty or not if set `:replace-empty-target?` to true. Need a blank? check here.